### PR TITLE
Adds in a security scan and also a license allowlist checker

### PR DESCRIPTION
**Pull request by michaelneale**

### CODE REVIEW

File: license-report.json [hunk 1]
```diff
@@ -0,0 +1 @@
...ps://www.mikealrogers.com/)"},{"department":"kessler","relatedTo":"stuff","name":"randombytes","licensePeriod":"perpetual","material":"material","licenseType":"MIT","link":"git+ssh://git@github.com/crypto-browserify/randombytes.git","remoteVersion":"2.1.0","installedVersion":"2.1.0","definedVersion":"2.1.0","author":""},{"department":"kessler","relatedTo":"stuff","name":"search-index","licensePeriod":"perpetual","material":"material","licenseType":"MIT","link":"git+https://github.com/fergiemcdowall/search-index.git","remoteVersion":"3.1.3","installedVersion":"3.1.3","definedVersion":"3.1.3","author":"Fergus McDowall"},{"department":"kessler","relatedTo":"stuff","name":"uuid","licensePeriod":"perpetual","material":"material","licenseType":"MIT","link":"git+https://github.com/uuidjs/uuid.git","remoteVersion":"8.3.2","installedVersion":"8.3.2","definedVersion":"8.3.2","author":"n/a"},{"department":"kessler","relatedTo":"stuff","name":"varint","licensePeriod":"perpetual","material":"material","licenseType":"MIT","link":"git://github.com/chrisdickinson/varint.git","remoteVersion":"6.0.0","installedVersion":"6.0.0","definedVersion":"6.0.0","author":"Chris Dickinson <chris@neversaw.us>"}]
```
Comment (thehenrytsai) [inline license-report.json:1]: Consider not checking in this and at the file to `.gitignore` instead.

File: package.json [hunk 2]
```diff
@@ -32,7 +32,8 @@
     "test:node": "npm run compile-validators && tsc && c8 mocha --es-module-specifier-resolution=node \"dist/esm/tests/**/*.spec.js\" && npm run make-coverage-badges",
     "test:node:ci": "npm run compile-validators && tsc && c8 mocha --es-module-specifier-resolution=node \"dist/esm/tests/**/*.spec.js\"",
     "test:browser": "npm run compile-validators && cross-env karma start karma.conf.cjs",
-    "test:browser-debug": "npm run compile-validators && cross-env karma start karma.conf.debug.cjs"
+    "test:browser-debug": "npm run compile-validators && cross-env karma start karma.conf.debug.cjs",
+    "license:check": "license-report --only=prod > license-report.json && node ./build/license-check.cjs"
```
Comment (thehenrytsai) [inline package.json:36]: Convention: use `license-check` instead, the `:` is used when there are variants of the same task

**Review by thehenrytsai — CHANGES_REQUESTED** [review]
Thanks Mic! Looks good overall, just some minor comments!

File: license-report.json (hunk 1 above)
Comment (michaelneale) [reply license-report.json:1]: yeah oops, didn't mean to push that in, thanks.

**Comment by michaelneale** [comment]
thanks @thehenrytsai - addressed

**Review by thehenrytsai — APPROVED** [review]

**Pull request merged**